### DISCUSSION TITLE
Can anybody show a block-wise transfer API example?

### DISCUSSION THREAD

**Discussion by souponaroom** [openthread/openthread#9002] [Q&A] [answered]
Hello, I have two devices, and would like them to communicate as follows:

1. Device A sends a GET to /request_block with empty payload
2. Device B sends a blockwise response to /request_block.

I see that there is a otCoapBlockwiseResource object to create, but I do not know how what each value does.

For example, if I receive a blockwise response in the above example, does "mReceiveHook" run, or does "mHandler" run? Or does "mReceiveHook" run every single time a new block is transferred?

Do I just set mTransmitHook to NULL since I am only expecting a blockwise response, but not a blockwise request? Is that even allowed, or is there a better way to do what I'm trying to do?

Thank you for your time.

**Comment by jwhui** [comment, marked as answer]
The OpenThread CLI is implemented on top of the OpenThread API and serves as an example of using the API.

You can find the CLI implementation of CoAP (including CoAP Block-Wise Transfers) in [src/cli/cli_coap.cpp](https://github.com/openthread/openthread/blob/main/src/cli/cli_coap.cpp).
